packages/assets-controllers/src/MultichainAssetsRatesController/MultichainAssetsRatesController.ts

```diff
@@ -417,14 +422,22 @@ export class MultichainAssetsRatesController extends StaticIntervalPollingContro
           'AccountsController:getSelectedMultichainAccount',
         );
       try {
-        const historicalPricesResponse = await this.#handleSnapRequest({
-          snapId: selectedAccount?.metadata.snap?.id as SnapId,
-          handler: HandlerType.OnAssetHistoricalPrice,
-          params: {
-            from: asset,
-            to: currentCaipCurrency,
+        const historicalPricesResponse = await this.messagingSystem.call(
+          'SnapController:handleRequest',
+          {
+            snapId: selectedAccount?.metadata.snap?.id as SnapId,
+            origin: 'metamask',
+            handler: HandlerType.OnAssetHistoricalPrice,
+            request: {
+              jsonrpc: '2.0',
+              method: HandlerType.OnAssetHistoricalPrice,
+              params: {
+                from: asset,
+                to: currentCaipCurrency,
+              },
+            },
```

Comment:
You shouldn't need to nest it like this. I would just revert back to just passing the params as you had before